### PR TITLE
Release v2.2.0

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
This release comes with some minor modifications to simplify the installation process and an important bugfix when subcycling is used (see #172).

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -22,13 +22,13 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine
+          pip3 install --break-system-packages setuptools wheel twine
```
Comment (IshaanDesai) [inline .github/workflows/build-and-test.yml:25]: Why not use pipx and do things cleanly here?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
           mkdir -p precice
           echo "from setuptools import setup" >> precice/setup.py
           echo "setup(name='pyprecice', version='3.0.0.0')" >> precice/setup.py
-          python3 -m pip install ./precice/
+          pip3 install --break-system-packages ./precice/
```
Comment (IshaanDesai) [inline .github/workflows/build-and-test.yml:40]: Same as above.

**Review by IshaanDesai — CHANGES_REQUESTED** [review]
Code related changes make sense; the CI stuff not so much.

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -22,13 +22,13 @@ jobs:
...
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine
+          pip3 install --break-system-packages setuptools wheel twine
       - name: Build distribution
         run:  python3 setup.py sdist
   test:
     name: Run mock unit tests
     runs-on: ubuntu-latest
-    container: quay.io/fenicsproject/stable:current
+    container: benjaminrodenberg/fenics
```
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply .github/workflows/build-and-test.yml:31]: Currently, we do not have a `precice/fenics` image and I did not want to get into the hassle creating one. `quay.io/fenicsproject/stable:current` does not work anymore. See https://github.com/precice/fenics-adapter/pull/174.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (BenjaminRodenberg) [reply .github/workflows/build-and-test.yml:25]: I have not used `pipx` a lot but I think `pipx` is not the right tool here. If I try it locally I get the following error:

```
$ pipx install setuptools

No apps associated with package setuptools or its dependencies. If you are
attempting to install a library, pipx should not be used. Consider using pip
or a similar tool instead.
```

I think the alternatives to `pip` are installation through the package manager, i.e., `apt install`, or going the `venv` path. Sure, we could do this, but I do not see a great benefit and for the sake of testing this approach is also good enough.

If you can suggest a one-liner change with `pipx`, ok. But I would rather not change more and keep it as it is. We can still change it with the next release.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (BenjaminRodenberg) [reply .github/workflows/build-and-test.yml:40]: Same as above :smirk:

**Comment by NiklasVin** [comment]
There are also two open PRs I created (#178 and #183) that close two issues (from 2020 and 2023). Don't we want to add them to the new release as well? If releasing this version is urgent, it makes sense to not include them.

File: .github/workflows/build-and-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (IshaanDesai) [reply .github/workflows/build-and-test.yml:31]: By not having the image in the precice account, I see a maintenance liability problem here which will hurt us in the future. But if it is too much work right now, we can keep this as it is.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (IshaanDesai) [reply .github/workflows/build-and-test.yml:25]: I tried to use `apt` to install the necessary build dependencies: https://github.com/precice/fenics-adapter/pull/185 but this does not seem to work. I do not want to invest too much time here, so lets go with what we have for now.

**Review by IshaanDesai — APPROVED** [review]
My earlier comments have been addressed to a reasonable degree. Release is good to go :+1:

File: .github/workflows/build-and-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply .github/workflows/build-and-test.yml:31]: Opened an issue: #186

**Pull request merged**